configure.in: fix cross compiling failed

The do_configure failed:
...
|checking for equivalent simple type of size_t... configure:
error: in `tmp/work/core2-64-poky-linux/socat/1.7.2.4-r0/build':
|configure: error: cannot run test program while cross compiling
...

For 1.7.2.4, the linux GCC check in configure.in didn't support
cross compiling, but the old version 1.7.2.3 supported, so we
reference 1.7.2.3 to fix this issue.

Upstream-Status: Inappropriate [oe specific]
Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 configure.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.in b/configure.in
--- a/configure.in
+++ b/configure.in
@@ -41,7 +41,7 @@ AC_CHECK_PROG(AR, ar, ar, gar)
 # fail
 AC_LANG_COMPILER_REQUIRE()
 
-if test "$CC" = "gcc"; then
+if test "$GCC" = yes; then
    CFLAGS="$CFLAGS -D_GNU_SOURCE -Wall -Wno-parentheses"
    ERRONWARN="-Werror -O0"
 elif test "$CC" = "clang"; then
-- 
1.8.1.2

